### PR TITLE
New Feature: Passcode Added ISSUE #132

### PR DESCRIPTION
**Pull request by shivamarora96**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 1]
```diff
@@ -147,7 +148,14 @@ public void onNext(Page<Client> clientPage) {
...
                         if (clientPage.getPageItems().size() != 0) {
                             long clientId = clientPage.getPageItems().get(0).getId();
+                            Context c = context.getApplicationContext();
+                            int mode = Context.MODE_PRIVATE;
+                            String spName = Constants.SHAREDPREF_NAME;
+                            SharedPreferences sp = c.getSharedPreferences(spName, mode);
                             getMvpView().showClient(clientId);
+                            String keyClientId = Constants.SP_KEY_CLIENTID;
+                            sp.edit().putLong(keyClientId, clientId);
+                            sp.edit().commit();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:158]: @shivamarora96 Please use utils/PreManager.class. don't need to make SharedPreferences object.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by therajanmaurya** [comment]
@shivamarora96 Please make above change and Squash your commits and force push again then i will look the logic of the implementation.

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java (hunk 1 above)
Comment (shivamarora96) [reply app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:158]: @therajanmaurya 

1.There is no such file named PreManager.class under util. Are you talking about api/local/PreferencesHelper?

2. **If you are talking about api/local/PreferencesHelper**, then what is the need for change ?

Comment (shivamarora96) [reply app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:158]: @therajanmaurya ?

**Comment by hardik124** [comment]
@therajanmaurya  can I take up this issue?

**Pull request closed without merging**